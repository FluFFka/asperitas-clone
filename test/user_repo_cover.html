
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>user_repo: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">asperitas-clone/pkg/user_repo/user_repo.go (100.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package user_repo

import (
        "crypto/md5"
        "database/sql"
        "encoding/hex"
        "errors"

        "asperitas-clone/pkg/items"

        _ "github.com/go-sql-driver/mysql"
)

type UserRepo struct {
        UserDB *sql.DB
}

func (repo *UserRepo) GetUserByID(id int) (*items.User, error) <span class="cov8" title="1">{
        user := &amp;items.User{}
        row := repo.UserDB.QueryRow("SELECT id, username, password FROM users WHERE id= ?", id)
        err := row.Scan(&amp;user.ID, &amp;user.Username, &amp;user.Password)
        if errors.Is(err, sql.ErrNoRows) </span><span class="cov8" title="1">{
                return nil, nil
        }</span> else<span class="cov8" title="1"> if err != nil </span><span class="cov8" title="1">{
                return nil, err
        }</span>
        <span class="cov8" title="1">return user, nil</span>
}

func (repo *UserRepo) GetUserByUsername(username string) (*items.User, error) <span class="cov8" title="1">{
        user := &amp;items.User{}
        row := repo.UserDB.QueryRow("SELECT id, username, password FROM users WHERE username= ?", username)
        err := row.Scan(&amp;user.ID, &amp;user.Username, &amp;user.Password)
        if errors.Is(err, sql.ErrNoRows) </span><span class="cov8" title="1">{
                return nil, nil
        }</span> else<span class="cov8" title="1"> if err != nil </span><span class="cov8" title="1">{
                return nil, err
        }</span>
        <span class="cov8" title="1">return user, nil</span>
}

func (repo *UserRepo) AddUser(user *items.User) (int, error) <span class="cov8" title="1">{
        var username string
        row := repo.UserDB.QueryRow("SELECT username FROM users WHERE username= ?", user.Username)
        err := row.Scan(&amp;username)
        if err == nil </span><span class="cov8" title="1">{
                return 0, items.ErrUserAlreadyExists
        }</span> else<span class="cov8" title="1"> if !errors.Is(err, sql.ErrNoRows) </span><span class="cov8" title="1">{
                return 0, err
        }</span>
        <span class="cov8" title="1">result, err := repo.UserDB.Exec(
                "INSERT INTO `users` (`username`, `password`) VALUES (?, ?)",
                user.Username,
                user.Password,
        )
        if err != nil </span><span class="cov8" title="1">{
                return 0, err
        }</span>
        <span class="cov8" title="1">id, err := result.LastInsertId()
        return int(id), err</span>
}

func HashPassword(password string) string <span class="cov8" title="1">{
        hashedPassword := md5.Sum([]byte(password))
        return hex.EncodeToString(hashedPassword[:])
}</span>

func (repo *UserRepo) Authorize(username, expPass string) (*items.User, error) <span class="cov8" title="1">{
        u, err := repo.GetUserByUsername(username)
        if err != nil </span><span class="cov8" title="1">{
                return nil, err
        }</span>
        <span class="cov8" title="1">if u == nil </span><span class="cov8" title="1">{
                return nil, items.ErrNoUser
        }</span>
        <span class="cov8" title="1">if HashPassword(expPass) == u.Password </span><span class="cov8" title="1">{
                return u, nil
        }</span>
        <span class="cov8" title="1">return nil, items.ErrBadPass</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
