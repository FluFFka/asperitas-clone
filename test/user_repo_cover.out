mode: set
asperitas-clone/pkg/user_repo/user_repo.go:18.64,22.35 4 1
asperitas-clone/pkg/user_repo/user_repo.go:27.2,27.18 1 1
asperitas-clone/pkg/user_repo/user_repo.go:22.35,24.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:24.8,24.23 1 1
asperitas-clone/pkg/user_repo/user_repo.go:24.23,26.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:30.79,34.35 4 1
asperitas-clone/pkg/user_repo/user_repo.go:39.2,39.18 1 1
asperitas-clone/pkg/user_repo/user_repo.go:34.35,36.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:36.8,36.23 1 1
asperitas-clone/pkg/user_repo/user_repo.go:36.23,38.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:42.62,46.16 4 1
asperitas-clone/pkg/user_repo/user_repo.go:51.2,56.16 2 1
asperitas-clone/pkg/user_repo/user_repo.go:59.2,60.21 2 1
asperitas-clone/pkg/user_repo/user_repo.go:46.16,48.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:48.8,48.43 1 1
asperitas-clone/pkg/user_repo/user_repo.go:48.43,50.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:56.16,58.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:63.43,66.2 2 1
asperitas-clone/pkg/user_repo/user_repo.go:68.80,70.16 2 1
asperitas-clone/pkg/user_repo/user_repo.go:73.2,73.14 1 1
asperitas-clone/pkg/user_repo/user_repo.go:76.2,76.41 1 1
asperitas-clone/pkg/user_repo/user_repo.go:79.2,79.30 1 1
asperitas-clone/pkg/user_repo/user_repo.go:70.16,72.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:73.14,75.3 1 1
asperitas-clone/pkg/user_repo/user_repo.go:76.41,78.3 1 1
